akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -168,6 +230,12 @@ private[http2] abstract class Http2Demux(http2Settings: Http2CommonSettings, ini
 
         // both client and server must send a settings frame as first frame
         multiplexer.pushControlFrame(SettingsFrame(initialLocalSettings))
+
+        if (pingState.isDefined) {
+          // to limit overhead rather than constantly rescheduling a timer and looking at system time we use a constant timer
```

Comment:
I think it might matter enough, because otherwise, we would have to cancel and reschedule the timer basically for every single incoming and outgoing frame.

Reply:
Ok, good, that was my thinking but it was just gut feeling which is why I left the comment.